src/coord/rmr/rq.c

```diff
@@ -181,21 +181,19 @@ static struct queueItem *rqPop(MRWorkQueue *q) {
 
     // Handle pending info logging. Access only to a non-NULL head and pendingInfo,
     // So it's safe to do without the lock.
-    const char *logLevel = "verbose";
-    if (q->head == q->pendingInfo.head) {
-      // If we hit the same head multiple times, we may have a problem. Increase log level.
-      if (++q->pendingInfo.hitCount > 100) logLevel = "notice";
+    if (q->head == q->pendingInfo.head && q->sz > q->pendingInfo.warnSize) {
+      // If we hit the same head multiple times, we may have a problem. Log it once.
+      RedisModule_Log(RSDummyContext, "warning", "Work queue at max pending with the same head. Size: %zu", q->sz);
+      q->pendingInfo.warnSize = q->sz + (1 << 10);
     } else {
       q->pendingInfo.head = q->head;
-      q->pendingInfo.hitCount = 1;
+      q->pendingInfo.warnSize = q->sz + (1 << 10);
```

Comment:
Why was this change done?
Also, can you add coverage?

Reply:
The previous implementation was too naive. A manual test I ran shows we get to this log too often. The goal of this log was to indicate we have a bug (if we mess the pending calculation, the RQ will become unresponsive)

Reply:
Coverage is only possible under heavy loads, which we don't have a way to replicate in our test suit today